### ISSUE TITLE
Wrong no-unsafe-member-access for reactive assignments and store subscriptions

### ISSUE THREAD

**Issue by dummdidumm** [sveltejs/eslint-plugin-svelte3#89]
#87 uncovered some limitations when using type-aware rules. Some of these are fixed by #88 , but not all of them. Reactive assignments and store subscriptions will fail:

```svelte
<script lang="ts">
  import { writable } from 'svelte/store';
  const store = writable([]);
  $store.length; // wrong no-unsafe-member-access error
  $: assignment = [];
  assignment.length; // wrong no-unsafe-member-access error
  // You can work around this by doing
  let another_assignment: string[];
  $: another_assignment = [];
  another_assignment.length; // OK
</script>
```

This is because the transformation just prepends generated `let X` statements above the code, with no specific type set to it. For it to work correctly, we would need to do transformations inline:

- `$: assignment = ..` --> `let assignment = ..` (but only if user did not declare the `let` himself)
- `const store = writable(...)` --> `const store = writable(..);let $store = get_store_value(store)` (where `get_store_value` is a fake function returning the `Value` in `Store<Value>`)

This would be similar to how we transform code in `svelte2tsx` for the intellisense features. The hard part is getting the line mappings correctly afterwards.

**Comment by dummdidumm** [comment]
This should be fixed for stores, for reactive statements it's not working yet.

**Comment by DetachHead** [comment]
is this the same issue?
```svelte
<script lang="ts">
    $: foo = 1
    foo.toString // no error
    const bar = () => {
        foo.toString // @typescript-eslint/no-unsafe-member-access
    }
</script>
```

**Comment by locmark** [comment]
> is this the same issue?
> 
> ```svelte
> <script lang="ts">
>     $: foo = 1
>     foo.toString // no error
>     const bar = () => {
>         foo.toString // @typescript-eslint/no-unsafe-member-access
>     }
> </script>
> ```

any updates on that? It's super annoying..

**Comment by wtachau** [comment]
any update?

**Comment by mohe2015** [comment]
The README of this project says it got replaced. Idk if that helps though

**Comment by wtachau** [comment]
Just noticed that :-/. I updated to `eslint-plugin-svelte`, which seems like it mostly works, though now I have to change some other configs. (For any future curious readers, I'm looking at https://github.com/sveltejs/eslint-plugin-svelte#parser-configuration and https://github.com/sveltejs/language-tools/issues/650)
